zio-http/src/main/scala/zio/http/URL.scala

```diff
@@ -98,6 +98,9 @@ final case class URL(
    * location.
    */
   def toJavaURL: Option[java.net.URL] = if (self.kind == URL.Location.Relative) None else Some(toJavaURI.toURL)
+
+  def queryParamsAsString: String =
+    self.queryParams.toList.map { case (k, v) => s"$k=${v.mkString(",")}" }.mkString("&")
```

Comment:
Each key and value has to be URL encoded, because they may have spaces, commas, `&`, and other characters inside them, which should not be interpreted as part of the structure of the query params.

I'd add a test to ensure they are URL encoded properly.

This also needs to be symmetric with the parsing of a URL. That is, if we parse a URL into `URL`, and then generate a string, that string needs to be equal to the original URL.

Perhaps you could use the `QueryStringEncoder` for this method that is used in `URL.encode` in order to ensure integrity between these different ways of producing the query param string.

Reply:
Changed the implementation to use code that has decoding built in. Changed the test to cover more cases.